tds/src/main/java/thredds/servlet/HtmlWriter.java

```diff
@@ -542,6 +552,84 @@ public int writeCatalog(HttpServletRequest req, HttpServletResponse res, InvCata
   }
 
   /**
+   * Parse markdownSourcePath as markdown and render to output as html
+   *
+   * @param output Appendable append the output to
+   * @param markdownSourcePath file path to markdown source
+   * @return true if markdownSourcePath is to a parsable file
+   */
+  private boolean generateMarkdown(Path markdownSourcePath, Appendable output) {
+    BufferedReader reader;
+    try {
+      reader = Files.newBufferedReader(markdownSourcePath, StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      return false;
+    } catch (java.lang.SecurityException e) {
+      return false;
+    }
+    Parser parser = Parser.builder().build();
+    Node node;
+    try {
+      node = parser.parseReader(reader);
+    } catch (IOException e) {
+      return false;
+    }
+    HtmlRenderer.builder().escapeHtml(true).build().render(node, output);
```

Comment:
Please note that HTML-tags are escaped, in order to keep the site maintainer in control, preventing html code injection.